File: Idempotent_key_redis/src/main/java/com/revy/example/api/TestController.java
package com.revy.example.api;

import com.revy.example.idempotency.annotion.Idempotent;
import io.swagger.v3.oas.annotations.Operation;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@RequestMapping("/api/v1")
public class TestController {

    @GetMapping
    @Idempotent
    @Operation(summary = "API 테스트", description = "멱등성 키를 포함하여 요청한다.")
    public ResponseEntity<String> test() {
        return ResponseEntity.ok("success");
    }

}

File: code-playbook/_kotlin/example_spring/kotlin_json/src/main/kotlin/com/revy/example/controller/TestController.kt
package com.revy.example.controller

import com.revy.example.controller.dto.TestRes
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping

@RestController
@RequestMapping("/")
class TestController {

    @PostMapping
    fun test():  TestRes {
        return TestRes(1L, "Test");
    }
}
